SalesforceSDK/Core/Settings/IApplicationInformationService.cs

```diff
@@ -33,7 +33,7 @@ public interface IApplicationInformationService
     {
         Task<string> GetApplicationDisplayNameAsync();
 
-        Task<string> GenerateUserAgentHeaderAsync(bool isHybrid);
+        Task<string> GenerateUserAgentHeaderAsync(bool isHybrid, bool isSmartSync);
```

Comment:
The SmartSync library is in its own separate library and it references the Core library(which is same as SalesforceSDK). The IApplicationInformationService and its implementation(which is in the Core library and has the UserAgent method) is called in the SmartSync library using the ServiceLocator pattern. I am going to change the UserAgent method to take in a string parameter/qualifier that gets appended to the user agent string.

Sounds good?